### PR TITLE
[GEOT-7063] New gt-cql2-text unsupported community module.

### PR DESCRIPTION
**Pull request by aaime**
[GEOT-7063](https://osgeo-org.atlassian.net/browse/GEOT-7063) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

First draft at implementing [CQL2](https://docs.ogc.org/DRAFTS/21-065.html). It is not complete, in particular:
- Time interval support works only with literals
- Extra interval time functions not present in ECQL are not yet included
- CASEI/ACCENTI are not implemented (ignoring accents is something new, will require several changes in stores as well)

That said, it can parse a good bunch of CQL2, and write it back. If you are curious about changes, see the tests, their javadocs contain a list of significant changes.

The CQL module has received a few changes to help extensibility.

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [ ] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [ ] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Review by dromagnoli — COMMENTED** [review]
In general, it looks good to me.
I have simply suggested some minor changes here and there to attempt same indentation or remove empty lines in long sequences of cases in switches that I have spotted out while "viewing" it. Nothing really important, just an esthetic touch :)

I have also spotted a long commented-out piece somewhere (see the review) that we might want to remove or just leave it there for other reasons (debug/reference/...)

Final question:
I have seen several classes containing a 2 in the name.
Are all of them new classes or do some of them are simply a rename or a refactor/move of older ones?

If all of them are new classes I think that we should update the copyright to 2021 since some of them still shows something like 2006-2008.

**Comment by aaime** [comment]
Most classes have a 2 because the language is called CQL2... or did I miss something? :-D

File: modules/unsupported/cql2/src/main/java/org/geootols/filter/text/cql_2/CQL2Compiler.java [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+                filter = this.builder.buildSpatialOverlapsFilter();
+                break;
+            default:
+                throw new CQLException("Binary spatial operator unexpected");
+        }
+
+        return filter;
+    }
+
+    //    private Filter buildBeforeOrDuring() throws CQLException {
```
[the remark this answers is not in the document]
Comment (aaime) [reply modules/unsupported/cql2/src/main/java/org/geootols/filter/text/cql_2/CQL2Compiler.java:495]: Forgot to remove it, thanks. The "Before or during" operator has no longer an equivalent in CQL2.

**Comment by dromagnoli** [comment]
> Most classes have a 2 because the language is called CQL2... or did I miss something? :-D

sorry. let me rephrase since I know that the classes with the 2 relates to CQL2 :-D
I was wondering if they are a copy and paste of some old-similar class from CQL or if they are brand new ones, in order to figure out the better YEAR to be put in the copyright header. (2022 vs 2008)

**Comment by aaime** [comment]
It's a bit of both... I'm going to check them one by one to make sure.

**Comment by aaime** [comment]
Cleaned up. Several classes are just copies indeed, mostly package-private classes. Left the header as is in them, they are supposed to go away when gt-cql2 and gt-cql get merged.

**Comment by dromagnoli** [comment]
+1
